components/scream/cmake/machine-files/perlmutter.cmake

```diff
@@ -1,11 +1,25 @@
 # Load all kokkos settings from Ekat's mach file
 set (EKAT_MACH_FILES_PATH ${CMAKE_CURRENT_LIST_DIR}/../../../../externals/ekat/cmake/machine-files)
-include (${EKAT_MACH_FILES_PATH}/kokkos/nvidia-a100.cmake)
-include (${EKAT_MACH_FILES_PATH}/kokkos/cuda.cmake)
+IF (USE_CUDA OR KOKKOS_ENABLE_CUDA)
```

Comment:
Does this actually work? IIRC, the machine file is loaded by cmake _before_ any command line option (the `-DBLAH=xyz`) are parsed. That means that there's no way to specify KOKKOS_ENABLE_CUDA in order to cause a switch in the mach file. At least _I don't think_ this might be a portable solution.

But I will have to check.

Reply:
I was lazy and used OR on both of those, but it looks like it's only USE_CUDA being set (so could remove the other)

For cime cases at least, with `--compiler=gnu`:
```
-- NDK Perlmutter USE_CUDA=FALSE KOKKOS_ENABLE_CUDA=
-- NDK Perlmutter CMAKE_CXX_COMPILER_ID=GNU CMAKE_Fortran_COMPILER_VERSION=10.3.0
```

and with `--compiler=gnugpu`:
```
-- NDK Perlmutter USE_CUDA=TRUE KOKKOS_ENABLE_CUDA=
-- NDK Perlmutter CMAKE_CXX_COMPILER_ID=GNU CMAKE_Fortran_COMPILER_VERSION=10.3.0
```

Reply:
Yes, for CIME cases it might be ok, since Macros.cmake sets USE_CUDA. I am not sure where those print statements were, but I tried on my machine to put a print of `CMAKE_Fortran_COMPILER_VERSION` right before and right after the call to `project` in `components/CMakeLists.txt`, and they print, respectively, "" and "8.5.0". Was the second print statement before the call to project? If so, then I'm puzzled.

Reply:
Those were in this file I'm updating `components/scream/cmake/machine-files/perlmutter.cmake`.  But I think your point is that we have (at least) 2 ways to build -- with cime and without. As I'm only testing with cime, might be problematic otherwise. Is there a way I can test on how it's being built within this file?  Like `IF (CIME_BUILD)`?  If so, does it make sense to use something like that here? I mean, in general, you can see what I'm trying to do here -- what's the best way?

Reply:
Yeah, exactly, we are doing 2 different things in standalone vs CIME builds.

If this is a CIME build, the mach file is loaded _before_ entering the scream dir, but _after_ the call to `project(E3SM...)`. So you could probably check the content of `PROJECT_NAME`. This should be "E3SM" for a CIME build, and empty for a standalone build (since we use the mach file as a cache to pre-load, which happens before scream's call to `project()`).

Reply:
I think what you are doing is fine. I also think we should prob switch to use the mach file in the same way that CIME does (i.e., including rather than using as a cache pre-load). I think for now checking PROJECT_NAME should be ok.

Reply:
I added a change